bolt-simple-taproot.md

```diff
@@ -0,0 +1,943 @@
+    * `to_remote_script` is the remote script:
+        ```
+        <remotepubkey> OP_CHECKSIGVERIFY 1 OP_CHECKSEQUENCEVERIFY
+        ```
+
+This output can be swept by the remote party with the following witness:
+```
+<to_remote_control_block> <to_remote_script> <remote_sig>
+```
+
+where `to_remote_control_block` is:
+```
+(parity_of_y(remotepubkey) | 0x0c) || <remotepubkey>
+```
+
+#### Anchor Outputs
+
+For simple taproot channels (`option_simple_taproot`) we'll maintain the same
+form as base segwit channels, but instead will utilize `local_delayedpubkey`
+and the `remotepubkey` rather than the multi-sig keys as that's no longer
+revealed due to musig2.
+
+An anchor output has the following form:
+
+  * `OP_1 anchor_output_key`
+  * where:
+    * `anchor_internal_key = remotepubkey/local_delayedpubkey`
+    * `anchor_output_key= anchor_internal_key + tagged_hash("TapTweak", anchor_internal_key || anchor_script_root)`
+    * `anchor_script = tapscript_root([anchor_script])`
+    * `anchor_script`:
+        ```
+        OP_16 OP_CHECKSEQUENCEVERIFY
+        ```
+
+### HTLC Scripts & Transactions
+
+#### Offered HTLCs
+
+We retain the same second-level HTLC mappings as base channels. The main
+modifications are using the revocation public key as the internal key, and
+splitting the timeout and success paths into individual script leaves.
+
+An offered HTLC has the following form:
+
+  * `OP_1 offered_htlc_key`
+  * where:
+    * `offered_htlc_key = revocation_pubkey + tagged_hash("TapTweak", revocation_pubkey || htlc_script_root)`
+    * `htlc_script_root = tapscript_root([htlc_timeout, htlc_success])`
+    * `htlc_timeout`:
+        ```
+        <local_htlcpubkey> OP_CHECKSIGADD <remote_htlcpubkey> OP_CHECKSIGADD 2 OP_EQUALVERIFY
+        1 OP_CHECKSEQUENCEVERIFY
+        ```
+    * `htlc_success`:
+        ```
+        OP_SIZE 32 OP_EQUALVERIFY OP_HASH160 <RIPEMD160(payment_hash)> OP_EQUALVERIFY
+        <remote_htlcpubkey>
+        OP_CHECKSIGVERIFY
+        1 OP_CHECKSEQUENCEVERIFY
+        ```
+
+In order to spend a offered HTLC, via either script path, an `inclusion_proof`
+must be specified along with the control block. This `inclusion_proof` is
+simply the `tap_leaf` hash of the path _not_ taken.
+
+#### Accepted HTLCs
+
+Accepted HTLCs inherit a similar format:
+
+  * `OP_1 accepted_htlc_key`
+  * where:
+    * `accepted_htlc_key = revocation_pubkey + tagged_hash("TapTweak", revocation_pubkey || htlc_script_root)`
+    * `htlc_script_root = tapscript_root([htlc_timeout, htlc_success])`
+    * `htlc_timeout`:
+        ```
+        <remote_htlcpubkey>
+        OP_CHECKSIGVERIFY
+        1 OP_CHECKSEQUENCEVERIFY OP_DROP
+        <cltv_expiry>
+        OP_CHECKLOCKTIMEVERIFY
+        ```
+    * `htlc_success`:
+        ```
+        OP_SIZE 32 OP_EQUALVERIFY OP_HASH160 <RIPEMD160(payment_hash)> OP_EQUALVERIFY
+        <local_htlcpubkey> OP_CHECKSIGADD <remote_htlcpubkey> OP_CHECKSIGADD 2 OP_EQUALVERIFY
```

Comment:
These need to be changed similarly to enforce the spending tx is the 2nd-level tx

Reply:
fixed in https://github.com/Roasbeef/lightning-rfc/pull/1